blocks/lists.ts

```diff
@@ -114,9 +116,15 @@ export const blocks = createBlockDefinitionsFromJsonArray([
   },
 ]);
 
-type CreateWithBlock = Block|typeof blocks['lists_create_with'];
 
-blocks['lists_create_with'] = {
+/** Type of a 'lists_create_with' block. */
+type CreateWithBlock = Block&ListCreateWithMixin;
+interface ListCreateWithMixin extends ListCreateWithMixinType {
+  itemCount_: number;
```

Comment:
This is because this is added in the `init` function? Could we just define it in the mixin instead of defining it in the constructor?

Here & below.

Reply:
I do believe that that is safe. We actually do that in the procedure blocks ([eg](https://github.com/google/blockly/blob/c688c9382b2878473b7072c20bd65ec40e27fe6b/blocks/procedures.js#L490)). All of the values get [mixed into the block](https://github.com/google/blockly/blob/c1e2956a318f459ffc4272b1fb9e115a2d9bb1a7/core/block.ts#L257) before the[ init method is called](https://github.com/google/blockly/blob/c1e2956a318f459ffc4272b1fb9e115a2d9bb1a7/core/block.ts#L264). So it's safe!